### ISSUE TITLE
Fails build on python3.11?

### ISSUE THREAD

**Issue by yuzibo** [google/python-fire#427]
Hi, 
I noticed the #1026619 that was marked removal for python-fire package on Debian.
The problem is that it build fails on here：
```python
I: pybuild base:240: cd /<<PKGBUILDDIR>>/.pybuild/cpython3_3.11/build; python3.11 -m unittest discover -v
fire.test_components (unittest.loader._FailedTest.fire.test_components) ... ERROR
fire.test_components_bin (unittest.loader._FailedTest.fire.test_components_bin) ... ERROR
fire.test_components_py3 (unittest.loader._FailedTest.fire.test_components_py3) ... ERROR
fire.test_components_test (unittest.loader._FailedTest.fire.test_components_test) ... ERROR
....
======================================================================
ERROR: fire.test_components (unittest.loader._FailedTest.fire.test_components)
----------------------------------------------------------------------
ImportError: Failed to import test module: fire.test_components
Traceback (most recent call last):
  File "/usr/lib/python3.11/unittest/loader.py", line 407, in _find_test_path
    module = self._get_module_from_name(name)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/unittest/loader.py", line 350, in _get_module_from_name
    __import__(name)
  File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.11/build/fire/test_components.py", line 28, in <module>
    from fire import test_components_py3 as py3  # pylint: disable=unused-import,no-name-in-module,g-import-not-at-top
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.11/build/fire/test_components_py3.py", line 58, in <module>
    class WithAsyncio(object):
  File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.11/build/fire/test_components_py3.py", line 60, in WithAsyncio
    @asyncio.coroutine
     ^^^^^^^^^^^^^^^^^
AttributeError: module 'asyncio' has no attribute 'coroutine'


======================================================================
ERROR: fire.test_components_bin (unittest.loader._FailedTest.fire.test_components_bin)
----------------------------------------------------------------------
ImportError: Failed to import test module: fire.test_components_bin
Traceback (most recent call last):
  File "/usr/lib/python3.11/unittest/loader.py", line 407, in _find_test_path
    module = self._get_module_from_name(name)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/unittest/loader.py", line 350, in _get_module_from_name
    __import__(name)
  File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.11/build/fire/test_components_bin.py", line 25, in <module>
    from fire import test_components
  File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.11/build/fire/test_components.py", line 28, in <module>
    from fire import test_components_py3 as py3  # pylint: disable=unused-import,no-name-in-module,g-import-not-at-top
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.11/build/fire/test_components_py3.py", line 58, in <module>
    class WithAsyncio(object):
  File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.11/build/fire/test_components_py3.py", line 60, in WithAsyncio
    @asyncio.coroutine
     ^^^^^^^^^^^^^^^^^
AttributeError: module 'asyncio' has no attribute 'coroutine'


======================================================================
ERROR: fire.test_components_py3 (unittest.loader._FailedTest.fire.test_components_py3)
----------------------------------------------------------------------
ImportError: Failed to import test module: fire.test_components_py3
Traceback (most recent call last):
  File "/usr/lib/python3.11/unittest/loader.py", line 407, in _find_test_path
    module = self._get_module_from_name(name)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/unittest/loader.py", line 350, in _get_module_from_name
    __import__(name)
  File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.11/build/fire/test_components_py3.py", line 58, in <module>
    class WithAsyncio(object):
  File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.11/build/fire/test_components_py3.py", line 60, in WithAsyncio
    @asyncio.coroutine
     ^^^^^^^^^^^^^^^^^
AttributeError: module 'asyncio' has no attribute 'coroutine'


======================================================================
ERROR: fire.test_components_test (unittest.loader._FailedTest.fire.test_components_test)
----------------------------------------------------------------------
ImportError: Failed to import test module: fire.test_components_test
Traceback (most recent call last):
  File "/usr/lib/python3.11/unittest/loader.py", line 407, in _find_test_path
    module = self._get_module_from_name(name)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/unittest/loader.py", line 350, in _get_module_from_name
    __import__(name)
  File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.11/build/fire/test_components_test.py", line 21, in <module>
    from fire import test_components as tc
  File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.11/build/fire/test_components.py", line 28, in <module>
    from fire import test_components_py3 as py3  # pylint: disable=unused-import,no-name-in-module,g-import-not-at-top
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.11/build/fire/test_components_py3.py", line 58, in <module>
    class WithAsyncio(object):
  File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.11/build/fire/test_components_py3.py", line 60, in WithAsyncio
    @asyncio.coroutine
     ^^^^^^^^^^^^^^^^^
AttributeError: module 'asyncio' has no attribute 'coroutine'


----------------------------------------------------------------------
Ran 7 tests in 0.001s
```

I have googled the asyncio was Deprecated [here](https://stackoverflow.com/questions/74343982/module-asyncio-has-no-attribute-coroutine-on-snakemake), right?
Could you have a look?

Thanks.

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1026619

**Comment by dbieber** [comment]
I've marked this as bug. We would like to support Python 3.11 and will need to update the asyncio code to do so.

**Comment by dvzrv** [comment]
Hi! A fix for this has been provided in #440 and it works.
Please merge and create a new release! We're currently rebuilding all packages against Python 3.11 on Arch Linux and I have to apply the mentioned fix to be able to Update this package.

Thanks! :)
